### PR TITLE
Add `CITATION.cff` and a method to automatically update this citation with new releases

### PR DESCRIPTION
**Pull request by Darylgolden**
## Overview: What does this pull request change?
GitHub recently added a [way to integrate citations directly with a repository](https://docs.github.com/en/github/creating-cloning-and-archiving-repositories/creating-a-repository-on-github/about-citation-files). This PR adds the file necessary for that feature to work. It also adds a function in our changelog generation script which will automatically update the version and release date when the script is run.

## Further Information and Comments
To test this, I created a new tag to simulate a new release so that I could test the changelog script. I should have removed it, but maybe check it just in case?

## Reviewer Checklist
- [x] The PR title is descriptive enough for the changelog, and the PR is labeled correctly
- [ ] If applicable: newly added non-private functions and classes have a docstring including a short summary and a PARAMETERS section
- [ ] If applicable: newly added functions and classes are tested

### CODE REVIEW

File: CITATION.cff [hunk 1]
```diff
@@ -0,0 +1,13 @@
+# YAML 1.2
+---
+authors: 
+  - 
+    name: "Manim Community Developers"
```
Comment (Darylgolden) [inline CITATION.cff:5]:
```suggestion
    name: "The Manim Community Developers"
```

**Review by ryanmccauley211 — APPROVED** [review]
LGTM from reading it, although I was trying to test it locally but not sure how. I think I need a github token for it

File: CITATION.cff [hunk 2]
```diff
@@ -0,0 +1,13 @@
+# YAML 1.2
+---
+authors: 
+  - 
+    name: "The Manim Community Developers"
+cff-version: "1.1.0"
+date-released: 2021-08-02
+license: MIT
+message: "We acknowledge the importance of good software to support research, and we note that research becomes more valuable when it is communicated effectively. To demonstrate the value of Manim, we ask that you cite Manim in your work."
+title: Manim
```
Comment (behackl) [inline CITATION.cff:10]: Quick question regarding the title: Just "Manim" vs. "Manim – Mathematical Animation Framework" (as it was in the bibtex snippet)?

**Review by behackl — APPROVED** [review]
LGTM in general, left one minor question. Can be merged as soon as resolved.

File: CITATION.cff (hunk 2 above)
Comment (Darylgolden) [reply CITATION.cff]: I've changed it to the second one.

**Pull request merged**